### DISCUSSION TITLE
Alias in prisma query

### DISCUSSION THREAD

**Discussion by Harsh070707** [prisma/prisma#22887] [Q&A]
### Question
```js
    mast_users_help_approval_requested_approved_by_userTomast_users: {
                select: {
                    user_id: true,
                    user_full_name: true
                }
            },
            mast_users_help_approval_requested_requested_byTomast_users: {
                select: {
                    user_id: true,
                    user_full_name: true
                }
            },
```
 i am getting this big field name = mast_users_help_approval_requested_requested_byTomast_users
 want to give it small name like as requested_by how I can give Prisma is not showing any option

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @Harsh070707 👋 

Thank you for rasing this question.

The feature to alias fields directly in a Prisma query is not currently supported. However, there are open feature request for this functionality as can be seen below.

- https://github.com/prisma/prisma/issues/4433
- https://github.com/prisma/prisma/issues/8151

For now, you would have to handle the renaming of the fields after you have fetched the data, using JavaScript's map function or similar methods to transform the data into the desired structure. 
```js
const data = await prisma.mast_users_help_approval_requested.findMany({
  select: {
    user_id: true,
    user_full_name: true,
    // ... other fields
  }
});

const transformedData = data.map(item => ({
  requested_by: {
    user_id: item.user_id,
    user_full_name: item.user_full_name
  },
  // ... map other fields as needed
}));
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by nikolasburk** [comment]
@Harsh070707 could you share the relevant parts of your Prisma schema please as well as the full Prisma Client query from your snippet above?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
